File: BE/src/main/java/com/example/Fstock/service/Service/CategoryService.java
package com.example.Fstock.service.Service;

import com.example.Fstock.dto.request.CreateCategoryRequest;
import com.example.Fstock.dto.response.CategoryDto;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CategoryService {
    List<CategoryDto> getAllCategories();
    CategoryDto getCategoryById(int id);
    CategoryDto createCategory(CreateCategoryRequest createCategoryRequest);
    Page<CategoryDto> getCategoriesPage(int pageNumber, int pageSize, String search);
    void deleteCategory(int categoryId);
    void updateCategory(int categoryId, CreateCategoryRequest createCategoryRequest);

}

File: FE/src/service/CategoryService.js
import api from "../api/api";
export const listCategories = () => {
    return api.get(`/categories`); 
    
}

export const getCategoryById = (id) => {
    return api.get(`/categories/${id}`);
}

export const categoriesInPage = (pageNumber, pageSize , search) => {
    return api.get(`/categories/page`, 
        {
        params: {
            pageNumber: pageNumber,
            pageSize: pageSize,
            search: search
        }

    }); 
    
}
export const createCategory = (category) => {
    return api.post(`/categories/create-category`, category);
}
export const deleteCategory = (categoryId) => {
    return api.delete(`/categories/delete/${categoryId}`);
}
export const updateCategory = (categoryId, data) => {
    return api.put(`/categories/update/${categoryId}`, data);
}
